src/index.ts

```diff
@@ -1,84 +1,90 @@
-const URL_LONG = /^(((http[s]?)|file):)?(\/\/)+([0-9a-zA-Z-_.=?&].+)$/
-const URL_SHORT = /^((\.|\.\.)?\/)([0-9a-zA-Z-_.=?&]+\/)*([0-9a-zA-Z-_.=?&]+)$/
-const isValidURL = (str: string) => URL_LONG.test(str) || URL_SHORT.test(str)
+const URL_LONG = /^(((http[s]?)|file):)?(\/\/)+([0-9a-zA-Z-_.=?&].+)$/;
+const URL_SHORT = /^((\.|\.\.)?\/)([0-9a-zA-Z-_.=?&]+\/)*([0-9a-zA-Z-_.=?&]+)$/;
+const isValidURL = (str: string) => URL_LONG.test(str) || URL_SHORT.test(str);
 
-export function createStyle (doc: Document, cssText: string) {
-  const style: HTMLStyleElement = doc.createElement('style')
+export function createStyle(doc: Document, cssText: string) {
+  const style: HTMLStyleElement = doc.createElement("style");
 
-  style.type = 'text/css'
-  style.appendChild(window.document.createTextNode(cssText))
+  style.type = "text/css";
+  style.appendChild(window.document.createTextNode(cssText));
 
-  return style
+  return style;
 }
 
-export function createLinkStyle (doc: Document, url: string) {
-  const style: HTMLLinkElement = doc.createElement('link')
+export function createLinkStyle(doc: Document, url: string) {
+  const style: HTMLLinkElement = doc.createElement("link");
 
-  style.type = 'text/css'
-  style.rel = 'stylesheet'
-  style.href = url
+  style.type = "text/css";
+  style.rel = "stylesheet";
+  style.href = url;
 
-  return style
+  return style;
 }
 
-export function createIFrame (parent: HTMLElement) {
-  const el: HTMLIFrameElement = window.document.createElement('iframe')
-  const css = 'visibility:hidden;width:0;height:0;position:absolute;z-index:-9999;bottom:0;'
+export function createIFrame(parent: HTMLElement) {
+  if (typeof window !== "undefined") {
+    const el: HTMLIFrameElement = window.document.createElement("iframe");
+    const css =
+      "visibility:hidden;width:0;height:0;position:absolute;z-index:-9999;bottom:0;";
 
-  el.setAttribute('src', 'about:blank')
-  el.setAttribute('style', css)
-  el.setAttribute('width', '0')
-  el.setAttribute('height', '0')
-  el.setAttribute('wmode', 'opaque')
+    el.setAttribute("src", "about:blank");
+    el.setAttribute("style", css);
+    el.setAttribute("width", "0");
+    el.setAttribute("height", "0");
+    el.setAttribute("wmode", "opaque");
 
-  parent.appendChild(el)
+    parent.appendChild(el);
 
-  return el
+    return el;
+  }
+  return null;
```

Comment:
**Second**, in the `createIFrame` function you are introducing a `null` return value. But I don't see anywhere you are covering that use case in the rest of the code below.

Please make sure that your new use case is covered by unit tests as well.

Reply:
here only return null if there is no window ... in my view there is window undefined case only when the building time.. when build is over then window will be there so ur code will work perfectly

Reply:
Sorry but you don't provide me concrete reasons. If you don't I think might try to adjusting your building process but out of the scope of this library.

Reply:
i think if anyone using ur same library in gatsby react . he will face the same problem which i face ...

problem is 

" Its will work in develop time , but when its comes to build face there is a problem windows not defined . that not mainly ur problem .. its about gatsby client side rendering thats why .. thats node doesnot have window ..proprieties .. Here i just use if condition for that rendering time problem .. if there is window undefined then its null ... 

i think this solution will help for who using gatsby like client side rendering frameworks .. if you think this PR is good .. then use otherwise leave it.... i just think if its help people just do it.. any way thanks for your reply